pdf/pdfdocument.cpp

```diff
@@ -162,6 +217,16 @@ void PDFDocument::requestPage(int index, int size, QQuickWindow *window, QRect s
     d->thread->queueJob(job);
 }
 
+void PDFDocument::requestArea(int index, int size, QQuickWindow *window, QRect subpart)
+{
+    if (!isLoaded() || isLocked() || subpart.isEmpty())
+        return;
+
+    RenderPageJob* job = new RenderPageJob(index, size, window, subpart,
+                                           RenderPageJob::RenderPageType::Area);
```

Comment:
This seems a bit silly that there are separate methods for requesting arbitrary areas of pages that are considered root texture and arbitrary areas that are considered patches. Could be nice if patch approach would be implementation detail of pdfcanvas. E.g. add some int extraData to requestPage and return that on finish signal.